Client/src/utils/sorting-algo.ts

```diff
@@ -0,0 +1,61 @@
+export function bubbleSortAlgo(arr: number[]) {
+  const ArrayStates: number[][] = [];
+  let animations:number[][] = []
+  let hasChanged = true;
+  while (hasChanged) {
+    hasChanged = false;
+    for (let i = 0; i <= arr.length - 2; i++) {
```

Comment:
< arr.length -1

Reply:
this can be Improved if every time you go throw the loop substract one to length because in the first iteration you make sure the biggest will be at the end, at the second the second biggest... and soo...

Reply:
`function bubbleSort(arr) {
  let swaps = false;
  let iterations = 0;
  while (swaps === false) {
        swaps = true;
    for (let i = 0; i < arr.length-1-iterations; i++) {
      if(arr[i]>arr[i+1]){
        swaps = false;
        [arr[i],arr[i+1]] =[arr[i+1],arr[i]];
      } 
    }
    iterations++
  }
}`

Reply:
This code take into account iterations... Much faster